libpod/container_inspect.go

```diff
@@ -0,0 +1,161 @@
+package libpod
+
+import (
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/projectatomic/libpod/libpod/driver"
+	"github.com/sirupsen/logrus"
+)
+
+func getContainerData(ctr *Container, size bool, driverData *driver.Data) (*ContainerData, error) {
+	config := ctr.config
+	runtimeInfo := ctr.state
+	spec := ctr.config.Spec
+
+	cpus, mems, period, quota, realtimePeriod, realtimeRuntime, shares := getCPUInfo(spec)
+	blkioWeight, blkioWeightDevice, blkioReadBps, blkioWriteBps, blkioReadIOPS, blkioeWriteIOPS := getBLKIOInfo(spec)
+	memKernel, memReservation, memSwap, memSwappiness, memDisableOOMKiller := getMemoryInfo(spec)
+	pidsLimit := getPidsInfo(spec)
+
+	args := config.Spec.Process.Args
+	path := args[0]
```

Comment:
We don't do any validation on the spec, so someone could feed us a bad spec with args having 0 length. Can you do a check for that here, to make sure we don't segfault trying to read an empty array?

Reply:
fixed